src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -117,37 +119,19 @@ public void acquireLease() throws ConnectException {
 
   @Override
   public void apply() throws ConnectException {
+    log.debug("Starting to apply WAL: {}", logFile);
+    if (!storage.exists(logFile)) {
+      log.debug("WAL file does not exist: {}", logFile);
+      return;
+    }
+    acquireLease();
+    log.debug("Lease acquired");
+
     try {
-      if (!storage.exists(logFile)) {
-        log.debug("Storage does not exist");
-        return;
-      }
-      acquireLease();
-      log.debug("Lease acquired");
       if (reader == null) {
         reader = new WALFile.Reader(conf.getHadoopConfiguration(), Reader.file(new Path(logFile)));
       }
-      Map<WALEntry, WALEntry> entries = new HashMap<>();
-      WALEntry key = new WALEntry();
-      WALEntry value = new WALEntry();
-      while (reader.next(key, value)) {
-        String keyName = key.getName();
-        if (keyName.equals(beginMarker)) {
-          entries.clear();
-        } else if (keyName.equals(endMarker)) {
-          for (Map.Entry<WALEntry, WALEntry> entry: entries.entrySet()) {
-            String tempFile = entry.getKey().getName();
-            String committedFile = entry.getValue().getName();
-            if (!storage.exists(committedFile)) {
-              storage.commit(tempFile, committedFile);
-            }
-          }
-        } else {
-          WALEntry mapKey = new WALEntry(key.getName());
-          WALEntry mapValue = new WALEntry(value.getName());
-          entries.put(mapKey, mapValue);
-        }
-      }
+      commitWalEntriesToStorage();
```

Comment:
this try might make more sense in the method

Reply:
the `WALFile.Reader` call (two lines above) throws one as well